submissions/maryOkosun/phpSlim/app/controllers/ArticleController.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace App\Controller;
+
+use DB\DB;
+use PDO;
+use PDOException;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class ArticleController
+{
+    private $db;
+
+    public function __construct()
+    {
+        $this->db = (new DB())->connect();
+    }
+
+    public function index(Request $request, Response $response, $args)
+    {
+        try {
+            $query = $this->db->prepare("select *
+            from articles
+            inner join admin
+            on articles.id = admin.id
+            where articles.created_by = admin.name;");
+            $query->execute();
+            $article = $query->fetchAll();
+            $response->getBody()->write(json_encode($article));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        } catch (PDOException $ex) {
+            $response->getBody()->write(json_encode(["error" => $ex->getMessage()]));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(500);
+        }
+    }
+
+    public function getArticle(Request $request, Response $response, $args)
+    {
+        $id = $args['id'];
+        try {
+            $query = $this->db->prepare("select *
+            from articles
+            inner join admin
+            on articles.id = admin.id
+            where articles.created_by = admin.name and articles.id = :id;");
+            $query->bindParam(":id", $id);
+            $query->execute();
+            $article = $query->fetch();
+
+            if (!$article) {
+                $response->getBody()->write(json_encode(["error" => "Article not found"]));
+                return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+            }
+
+            $response->getBody()->write(json_encode($article));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        } catch (PDOException $ex) {
+            $response->getBody()->write(json_encode(["error" => $ex->getMessage()]));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(500);
+        }
+    }
+
+    public function create(Request $request, Response $response, $args)
+    {
+        $requestData = $request->getParsedBody();
+        $title = $requestData['title'];
+        $description = $requestData['description'];
+        $status = $requestData['status'];
```

Comment:
Status should have a default value of `unpublished`, it shouldn't be pass as a post data here. The default value should be set when creating your `articles` table

Reply:
I have effected all the corrections you made. Please review. Thanks